components/blocks/FilesToCards.vue

```diff
@@ -59,26 +61,229 @@
+                    ><i class="mdi mdi-link-variant" /></span></span
+              ></a>
+              <a
+                v-if="item.links.documentation"
+                class="link-item d-button has-text-link"
+                :href="item.links.documentation"
+                ><span
+                  >DOCUMENTATION<span class="icon ml-2"
+                    ><i class="mdi mdi-book" /></span></span
+              ></a>
+              <a
+                v-if="item.links.publication"
+                class="link-item d-button has-text-link"
+                :href="item.links.publication"
+                ><span
+                  >PUBLICATION<span class="icon ml-2"
+                    ><i class="mdi mdi-newspaper" /></span></span
+              ></a>
+              <a
+                v-if="item.links.doi"
+                class="link-item d-button has-text-link"
+                :href="'https://' + item.links.doi"
+                ><span
+                  >PUBLICATION<span class="icon ml-2"
+                    ><i class="mdi mdi-newspaper" /></span></span
+              ></a>
+            </div>
+          </template>
+        </DCard>
+      </div>
+    </div>
   </main>
 </template>
 
 <script>
 import DCard from '@/components/base/DCard.vue';
+import Multiselect from 'vue-multiselect';
 
 export default {
   components: {
     DCard,
+    Multiselect,
   },
   props: {
     data: {
       type: Array,
       required: true,
     },
   },
+  data: () => ({
+    ascending: true,
+    sortBy: [],
+    searchGroup: [],
+    sortOptions: [],
+  }),
   computed: {
     filteredData() {
       return this.data.filter((d) => !d.hidden);
     },
+    sortedArray() {
+      let tempCards = this.data;
+
+      // Filter by group tag
+      let filtered = tempCards;
+      if (this.searchGroup.length > 0) {
+        filtered = [];
+        for (const arr in tempCards) {
+          for (const filter in this.searchGroup) {
+            if (tempCards[arr].group === this.searchGroup[filter]) {
+              filtered.push(tempCards[arr]);
+            }
+          }
+        }
+      }
+
+      // Sort by alphabetical order
+      if (this.sortBy.name === 'Title') {
+        tempCards = filtered.sort((a, b) => {
+          const fa = a.title.toLowerCase();
+          const fb = b.title.toLowerCase();
+          if (fa < fb) {
+            return -1;
+          }
+          if (fa > fb) {
+            return 1;
+          }
+          return 0;
+        });
+      } else if (this.sortBy.name === 'Date') {
+        // Sort by date
+        tempCards = filtered.sort((a, b) => {
+          return new Date(a.date) - new Date(b.date);
+        });
+      }
+
+      // Show sorted array in descending or ascending order
+      if (!this.ascending) {
+        filtered.reverse();
+      }
+
+      return filtered;
+    },
+  },
+  methods: {
+    cardTags() {
+      const tags = [];
+      this.data.forEach(function (obj) {
```

Comment:
make this a map instead and chain into the filter and sort
```js
const tags = this.data.map((card) => card.group)
return tags.filter((group, index) => tags.indexOf(group) === index).sort();
```

Generally, we prefer using fat arrow functions instead of es5 functions (unless the scope of `this` becomes a problem)